src/main/java/edu/kis/powp/jobs2d/canva/visitor/CommandWithinShapeVisitor.java

```diff
@@ -0,0 +1,71 @@
+package edu.kis.powp.jobs2d.canva.visitor;
+
+import edu.kis.powp.jobs2d.canva.shapes.CanvaShape;
+import edu.kis.powp.jobs2d.command.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CommandWithinShapeVisitor implements DriverCommandVisitor {
+    private final CanvaShape canvaShape;
+    private final List<Boolean> containsList;
+
+    public CommandWithinShapeVisitor(CanvaShape canvaShape) {
+        this.canvaShape = canvaShape;
+        this.containsList = new ArrayList<>();
+    }
+
+    @Override
+    public void visit(SetPositionCommand command) {
+        boolean result = canvaShape.contains(command.getPosX(), command.getPosY());
+        containsList.add(result);
+    }
+
+    @Override
+    public void visit(OperateToCommand command) {
+        boolean result = canvaShape.contains(command.getPosX(), command.getPosY());
+        containsList.add(result);
+    }
+
+    @Override
+    public void visit(ICompoundCommand command) {
+        for (DriverCommand subCommand : command) {
+            subCommand.accept(this);
+        }
```

Comment:
OK, but this is not a stateless aproach (requires reset() between visitor calls)